### PR TITLE
Add reporting/v1 service

### PR DESCRIPTION
**Pull request by johanstokking**
#### Summary

Add `reporting/v1` service

#### Changes

This adds a new service, `org.packetbroker.reporting.v1.Reporter`, for getting routed messages.

This serivce may be expanded with other types of reports, like networks seen and observed gateway RF parameters.

### CODE REVIEW

File: reporting/v1/service.proto [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  // LoRa Alliance NetID of the Forwarder Member.
+  uint32 forwarder_net_id = 2;
+  // Tenant ID managed by the Forwarder Member.
+  string forwarder_tenant_id = 3;
+  // LoRa Alliance NetID of the Home Network Member.
+  uint32 home_network_net_id = 4;
+  // Tenant ID managed by the Home Network Member.
+  string home_network_tenant_id = 5;
+  // PHYPayload type.
+  PHYPayloadType phy_payload_type = 6;
```
Comment (htdvisser) [inline reporting/v1/service.proto:81]: Why not add counter fields for each type separately?

File: reporting/v1/service.proto [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  // Count of all messages that are routed by Packet Broker.
+  uint64 routed_count = 7;
+  // Count of all messages processed successfully by the target.
+  uint64 processed_success_count = 8;
+  // Count of all messages processed with an error reported by the target.
+  // For uplink messages, the key corresponds to org.packetbroker.v3.UplinkMessageProcessingError.
+  // For downlink messages, the key corresponds to org.packetbroker.v3.DownlinkMessageProcessingError.
+  map<uint32, uint64> processed_error_count = 9;
```
Comment (htdvisser) [inline reporting/v1/service.proto:89]: Even with the comment, it's difficult to understand what's happening here.
Wouldn't it be more intuitive to have something like this?

```proto
message RoutedMessagesRecord {
	// ...
	
	message UplinkErrorCount {
		org.packetbroker.v3.UplinkMessageProcessingError error_type = 1;
		uint64 count = 2;
	}
	  
	message DownlinkErrorCount {
		org.packetbroker.v3.DownlinkMessageProcessingError error_type = 1;
		uint64 count = 2;
	}

	oneof processed_error_count { // oneof not needed if we add fields for each type (see review comment above)
		repeated UplinkErrorCount processed_uplink_error_count = 9;
		repeated DownlinkErrorCount processed_downlink_error_count = 10;
	}
}
```

File: reporting/v1/service.proto [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    message ErrorCount {
+      org.packetbroker.v3.UplinkMessageProcessingError error_type = 1;
+      uint64 count = 2;
+    }
+    // Count of join-requests that are routed by Packet Broker.
+    uint64 join_requests_routed_count = 1;
+    // Count of join-requests processed successfully by the Home Network.
+    uint64 join_requests_processed_success_count = 2;
+    // Count of join-requests processed with an error by the Home Network.
+    ErrorCount join_requests_processed_error_count = 3;
```
Comment (htdvisser) [inline reporting/v1/service.proto:86]: This should be repeated, right? Same for the other ones

Comment (johanstokking) [reply reporting/v1/service.proto:86]: 🤦‍♂️

**Pull request merged**